inferno-vc/src/Inferno/VersionControl/Server.hs

```diff
@@ -125,58 +120,45 @@ vcServer toHandler =
       Map.fromList <$> (forM hs $ \h -> (h,) <$> Ops.fetchVCObject h)
 
 runServer ::
-  forall proxy m a g env config.
+  forall m env config.
   ( HasField "serverHost" config config T.Text T.Text,
     HasField "serverPort" config config Int Int,
-    VCHashUpdate a,
-    VCHashUpdate g,
+    VCHashUpdate (Ops.Author m),
+    VCHashUpdate (Ops.Group m),
     FromJSON config,
-    FromJSON a,
-    FromJSON g,
-    ToJSON a,
-    ToJSON g,
-    Ops.Deserializable m a,
-    Ops.Deserializable m g,
-    Ops.Serializable m a,
-    Ops.Serializable m g,
-    Ops.InfernoVCOperations VCServerError m,
-    Ord g
+    FromJSON (Ops.Author m),
+    FromJSON (Ops.Group m),
+    ToJSON (Ops.Author m),
+    ToJSON (Ops.Group m),
+    Ops.InfernoVCOperations VCServerError m
   ) =>
-  proxy a ->
-  proxy g ->
   (forall x. config -> IOTracer T.Text -> (env -> IO x) -> IO x) ->
   (forall x. m x -> env -> ExceptT VCServerError IO x) ->
   IO ()
-runServer proxyA proxyG withEnv runOp = do
+runServer withEnv runOp = do
   readServerConfig "config.yml" >>= \case
     Left err -> putStrLn err
-    Right serverConfig -> runServerConfig proxyA proxyG withEnv runOp serverConfig
+    Right serverConfig -> runServerConfig withEnv runOp serverConfig
 
 runServerConfig ::
-  forall proxy m a g env config.
+  forall m env config.
```

Comment:
`a` and `g` can now be inferred from `m` thanks to the `InfernoVCOperations` constraint so we don't need no proxies